modules/Microsoft.CDN/profiles/endpoints/deploy.bicep

```diff
@@ -32,36 +32,34 @@ resource defaultTelemetry 'Microsoft.Resources/deployments@2021-04-01' = if (ena
   }
 }
 
-resource profileName_endpoint 'microsoft.cdn/profiles/endpoints@2021-06-01' = if (!empty(endpointName)){
+resource endpoint 'microsoft.cdn/profiles/endpoints@2021-06-01' = if (!empty(endpointName)) {
   parent: profile
   name: endpointName
   location: location
   properties: endpointProperties
   tags: tags
 }
 
-resource profile_EndpointOrigins 'Microsoft.Cdn/profiles/endpoints/origins@2021-06-01' = if (!empty(endpointProperties.origins)){
-  parent: profileName_endpoint
-  name: endpointProperties.origins[0].name
-  properties: {
-    hostName: endpointProperties.origins[0].properties.hostName
-    httpPort: endpointProperties.origins[0].properties.httpPort
-    enabled: endpointProperties.origins[0].properties.enabled
-    httpsPort: endpointProperties.origins[0].properties.httpsPort
+module profile_EndpointOrigin 'origins/deploy.bicep' = {
+  name: '${endpointName}-origins'
+  params: {
+    profileName: profile.name
+    endpointName: endpointName
+    originsProperties: endpointProperties.origins
   }
 }
 
```

Comment:
Can there be an origin without an endpoint? 

Why is there a condition `if (!empty(endpointName))`?. I don't think it should be optional to deploy an endpoint in the endpoint module - but rather make it optional in its parent.
Also, I'd understand if the origin is optional - unless it's required of course.

Reply:
No - This was a misconfiguration of the file - I have updated the condition and added a for loop for origins in latest commit